File: scratch-backend/test_api.py
import urllib.request
import json

url = "http://localhost:5000/api/v1/ai/recommend-routine"
data = json.dumps({"skinType": "Dry", "concerns": ["dry_flaky"]}).encode("utf-8")
headers = {"Content-Type": "application/json"}

req = urllib.request.Request(url, data=data, headers=headers, method="POST")
try:
    with urllib.request.urlopen(req) as res:
        print(json.dumps(json.loads(res.read().decode("utf-8")), indent=2))
except Exception as e:
    print("Error:", e)

File: scratch-backend/test_api.js
const http = require('http');

const data = JSON.stringify({
  skinType: 'Dry',
  concerns: ['dry_flaky']
});

const options = {
  hostname: 'localhost',
  port: 5000,
  path: '/api/v1/ai/recommend-routine',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const req = http.request(options, res => {
  let body = '';
  res.on('data', chunk => body += chunk);
  res.on('end', () => console.log(JSON.stringify(JSON.parse(body), null, 2)));
});

req.on('error', error => console.error(error));
req.write(data);
req.end();
